Framework/Algorithms/src/MaxEnt/MaxentCalculator.cpp

```diff
@@ -292,17 +306,27 @@ void MaxentCalculator::calculateChisq() {
   if (m_data.empty() || m_errors.empty() || m_dataCalc.empty()) {
     throw std::runtime_error("Cannot calculate chi-square");
   }
+
+  // Calculate
+  // ChiSq = sum_i [ data_i - dataCalc_i ]^2 / [ error_i ]^2
+  m_chisq = calculateChiSquared(m_dataCalc);
+}
+
+double
+MaxentCalculator::calculateChiSquared(const std::vector<double> &data) const {
   size_t npoints = m_data.size();
+  auto dpoints = static_cast<double>(npoints);
 
   // Calculate
   // ChiSq = sum_i [ data_i - dataCalc_i ]^2 / [ error_i ]^2
-  m_chisq = 0;
+  double chisq = 0;
   for (size_t i = 0; i < npoints; i++) {
     if (m_errors[i] != 0.0) {
```

Comment:
You did not add "if (m_errors[i] != 0.0)" but probably it was added for a reason (unfortunately no comment). However with that if statement you may be dividing by the wrong number! Could you please look into understanding what this is doing and add a comment to explain why it is there (and if not remove). If the if statement is needed then the number of data points equals the number of times the content of this if statement is called.

Also please change the method description to include the 1/N

Reply:
I just noticed that dividing every term by `dpoints` is very inefficient.

I can see the reason for the `if` in protecting against accidental (practically impossible) zero errors. 
I am afraid that if I change `dpoints` to the number of non-zero errors it might affect the code that calculates `c1, c2, s1, s2` values. I don't see benefits in trying to do it because:
1. probability of having significant number of zeros is very low
2. inaccuracy in the `chi2` can be offset at run-time by adjusting the target chi property.